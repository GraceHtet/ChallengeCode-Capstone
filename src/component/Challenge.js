import { useParams, useNavigate } from 'react-router-dom';
import bArrow from '../assets/img/backArrow.svg';

const Challenge = () => {
  const { name } = useParams();
  const navigate = useNavigate();

  const styles = {
    background: 'none',
    border: 'none',
    cursor: 'pointer',
  };

  return (
    <>
      <button type="button" style={styles} onClick={() => navigate(-1)}>
        <img src={bArrow} width="30px" alt="backArrow" />
      </button>
      <p>{name}</p>
    </>
  );
};

export default Challenge;
