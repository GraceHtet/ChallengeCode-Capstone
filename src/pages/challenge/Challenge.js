import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { fetchChallenges } from '../../redux/challenges/challengesSlice';
import Img from '../../component/Img';
import Navbar from '../../component/Navbar';
import styles from './Challenge.module.css';

const Challenge = () => {
  const { name } = useParams();
  const { challenges } = useSelector((state) => state.challenges);
  const dispatch = useDispatch();

  useEffect(() => {
    if (challenges.siteName !== name) dispatch(fetchChallenges(name));
  }, [challenges, dispatch]);

  const upcomings = challenges.filter((each) => {
    const cMonth = new Date(each.startTime).getMonth();
    const cyear = new Date(each.startTime).getFullYear();
    const curMonth = new Date().getMonth();
    const curYear = new Date().getFullYear();
    return cMonth > curMonth && cyear >= curYear;
  });

  return (
    <>
      <Navbar back title="Upcoming/Challenges" />

      <div className={styles.container}>
        <Img siteName={name} width="90%" height="70px" />
        <h2 className={styles.h2}>{name.toUpperCase().replace('_', ' ')}</h2>
      </div>
      {upcomings.map((each) => (
        <div key={crypto.randomUUID()} className={styles.div}>
          <p className={styles.title}>{each.name}</p>
          <p>
            duration -{each.duration}
            hr
          </p>
          <p>
            Start -{new Date(each.startTime).getFullYear()}/{new Date(each.startTime).getMonth()}/
            {new Date(each.startTime).getDate()}
          </p>
          <p>
            End -{new Date(each.endTime).getFullYear()}/{new Date(each.endTime).getMonth()}/
            {new Date(each.endTime).getDate()}
          </p>
          <a href={each.url} className={styles.a}>
            Go to Challenge
            <svg
              width="16px"
              height="16px"
              viewBox="0 0 20 20"
              className={styles.svg}
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g id="SVGRepo_bgCarrier" strokeWidth="0" />
              <g id="SVGRepo_tracerCarrier" strokeLinecap="round" strokeLinejoin="round" />
              <g id="SVGRepo_iconCarrier">
                {' '}
                <path
                  d="M4 12L20 12L14.0001 18M17 9L14 6"
                  stroke="#000000"
                  stroke-width="2.088"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                />{' '}
              </g>
            </svg>
          </a>
        </div>
      ))}
    </>
  );
};

export default Challenge;
