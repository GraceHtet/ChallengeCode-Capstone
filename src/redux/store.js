import { configureStore } from '@reduxjs/toolkit';
import challengesReducer from './challenges/challengesSlice';
import challengeReducer from './challenge/challengeSlice';

const store = configureStore({
  reducer: {
    challenges: challengesReducer,
    challenge: challengeReducer,
  },
});

export default store;
