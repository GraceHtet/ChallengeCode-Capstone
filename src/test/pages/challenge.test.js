import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import store from '../../redux/store';
import Challenge from '../../pages/challenge/Challenge';

describe('Challenge', () => {
  test('Render challenge page', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Challenge />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
