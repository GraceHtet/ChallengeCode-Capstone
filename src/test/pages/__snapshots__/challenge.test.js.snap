// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Challenge Render challenge page 1`] = `
Array [
  <nav
    className="nav"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="backArrow"
        src="backArrow.svg"
        width="20px"
      />
    </button>
    <span>
      Upcoming/Challenges
    </span>
    <svg
      className="svg"
      fill="#55544e"
      height="20px"
      id="Layer_1"
      stroke="#9b7a03"
      version="1.1"
      viewBox="0 0 512 512"
      width="20px"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <g
        id="SVGRepo_bgCarrier"
        strokeWidth="0"
      />
      <g
        id="SVGRepo_tracerCarrier"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <g
        id="SVGRepo_iconCarrier"
      >
         
        <g>
           
          <g>
             
            <path
              d="M511.956,308.448L512,206.866l-65.97-7.239c-3.584-12.101-8.323-23.822-14.165-35.037l42.198-52.531l-71.812-71.845 L350.48,81.766c-11.115-6.041-22.751-10.987-34.783-14.783l-7.318-66.961H206.797l-7.21,65.973 c-11.988,3.556-23.608,8.248-34.726,14.021l-52.475-42.265l-71.938,71.72l41.484,51.825c-6.058,11.109-11.02,22.741-14.831,34.763 L0.134,203.29L0,304.872l65.963,7.295c3.573,12.101,8.301,23.826,14.135,35.049L37.856,399.71l71.751,71.907l51.807-41.507 c11.112,6.052,22.744,11.008,34.769,14.815l7.261,66.966l101.582,0.088l7.266-65.967c12.1-3.578,23.826-8.313,35.043-14.149 l52.513,42.22l71.876-71.783l-41.529-51.788c6.046-11.112,10.997-22.747,14.8-34.777L511.956,308.448z M256.021,347.705 c-50.659,0-91.727-41.068-91.727-91.727s41.068-91.727,91.727-91.727c50.659,0,91.727,41.068,91.727,91.727 S306.681,347.705,256.021,347.705z"
            />
             
          </g>
           
        </g>
         
      </g>
    </svg>
  </nav>,
  <div
    className="container"
  >
    <img
      alt=""
      height="70px"
      src="pageImg.svg"
      width="90%"
    />
    <h2
      className="h2"
    />
  </div>,
]
`;
